### ISSUE TITLE
Mock service for unit tests

### ISSUE THREAD

**Issue by ghost** [ntegral/nestjs-sendgrid#15]
Hi, Great package.

I am having some trouble mocking the service for my unit tests. It seems like, if I provide the SendGridModule in the imports it ignores the provider override and actually sends the email. If I do not provide the SendGridModule this is the error

`Nest can't resolve dependencies of the MailerService (?, ConfigService). Please make sure that the argument SendGridToken at index [0] is available in the RootTestModule context.`

```ts
const mockSendGridService = {
  send: jest.fn().mockResolvedValue(null)
};

const mockConfigService = {
  get: jest.fn().mockReturnValue('')
};

describe('MailerService', () => {
  let service: MailerService;

  beforeEach(async () => {
    const module: TestingModule = await Test.createTestingModule({
      providers: [
        MailerService,
        { provide: SendGridService, useValue: mockSendGridService },
        { provide: ConfigService, useValue: mockConfigService }
      ]
    }).compile();

    service = module.get<MailerService>(MailerService);
  });

  it('should be defined', () => {
    expect(service).toBeDefined();
  });
});
```

can you tell me where im going wrong?

Thanks

**Comment by jquinteros-applaudo** [comment]
Hey, you need to pass `SENDGRID_TOKEN ` const from `nestjs-sendgrid ` package in the `provide `property when creating your testing module. And just in case, pass the same value when getting the provider inside the `beforeEach` function

tl;dr -> The DI identifier/token for the `SendGrid` service is not the class, it's the constant `SENDGRID_TOKEN `
